torch/distributed/algorithms/join.py

```diff
@@ -7,15 +7,15 @@
 import torch.distributed as dist
 
 
-class _JoinHook():
+class JoinHook():
     r"""
     This defines a join hook, which provides two entry points in the join
     context manager: a main hook, which is called repeatedly while there exists
     a non-joined process, and a post-hook, which is called once all processes
     have joined.
 
     To implement a join hook for the generic join context manager, define a
-    class that inherits from :class:`_JoinHook`, override ``main_hook()`` and
+    class that inherits from :class:`JoinHook`, override ``main_hook()`` and
     ``post_hook()`` as appropriate, and override ``device()`` and
```

Comment:
`device()` and `process_group()` methods are not available in `JoinHook`. Do you mean `join_device()` and `join_process_group()` in `Joinable`?

Reply:
Good catch. This is leftover from when `device` and `process_group` were part of `JoinHook`. I will fix this.